index.html

```diff
@@ -1,453 +1,585 @@
+            <li><code>string</code> - maps to <code>string</code></li>
+          </ol>
+        </li>
+        <li>If the type of the value is an array, ensure each element of the array has the same primitive type. Push an
+          object to <code>types</code> with the <code>name</code> set to the property name of the input document and
+          <code>type</code> set to the corresponding EIP712 array type
+          <ol class="algorithm">
+            <li><code>boolean[]</code> - maps to <code>bool[]</code></li>
+            <li><code>number[]</code> - maps to <code>uint256[]</code></li>
+            <li><code>string[]</code> - maps to <code>string[]</code></li>
+          </ol>
+          <p class="example" title="Arrays of structs are not yet supported">
+            WARNING: The current algorithm definition does not support auto generating types for arrays of structs. We
+            need
+            to
+            work on that.
+          </p>
+        </li>
+        <li>If the type of the value is an object, call the function recursively on the inner object, and set the return
+          value equal to <code>_recursiveOutput</code>.
+          <ol class="algorithm">
+
+            <li>Set <code>_recursiveTypes = _recursiveOutput[primaryType]</code></li>
+            <li>Push an object to <code>types</code> with the <code>name</code> set to the property name of the input
+              document and <code>type</code> set to the CapitalCased property name - <code>propertyType</code>
+            </li>
+            <li>
+              Set <code>output[propertyType] = _recursiveTypes</code>
+            </li>
+            <li>Loop over <code>_recursiveOutput</code>, and if any keys other than <code>primaryType</code> are
+              present,
+              add them directly to <code>output</code></li>
+          </ol>
+        </li>
 
-    </section>
+        <li>Finally, set <code>output[primaryType]</code> to the <code>types</code> array that was generated. Return
+          <code>output</code>
+        </li>
+      </ol>
 
-    <section id="terminology">
-      <h2>Terminology</h2>
-      <p>
-        The following terms are used to describe concepts involved in the
-        generation and verification of the Linked Data Proof
-        <a>signature suite</a>.
-      </p>
+      The following is an example of the autogenerated schema. Given the following input document and no `primaryType`:
+      <pre class="example">
+          {
+            "@context": ["https://schema.org", "https://w3id.org/security/v2"],
+            "@type": "Person",
+            name: {
+              first: "Jane",
+              last: "Doe",
+            },
+            otherData: {
+              jobTitle: "Professor",
+              school: "University of ExampleLand",
+            },
+            telephone: "(425) 123-4567",
+            email: "jane.doe@example.com",
+          };
```

Comment:
Quoting the property names and removing the trailing semicolon would improve consistency, and enable using the data more easily in places that expect pure JSON:
```suggestion
            "name": {
              "first": "Jane",
              "last": "Doe",
            },
            "otherData": {
              "jobTitle": "Professor",
              "school": "University of ExampleLand",
            },
            "telephone": "(425) 123-4567",
            "email": "jane.doe@example.com",
          }
```